pkg/operator/staticpod/controller/guard/guard_controller.go

```diff
@@ -280,9 +289,13 @@ func (c *GuardController) sync(ctx context.Context, syncCtx factory.SyncContext)
 	return utilerrors.NewAggregate(errs)
 }
 
-func IsSNOCheckFnc(infraLister configv1listers.InfrastructureLister) func() (bool, error) {
+// IsSNOCheckFnc creates a function that checks if the topology is SNO
+func IsSNOCheckFnc(infraInformer configv1informers.InfrastructureInformer) func() (bool, error) {
 	return func() (bool, error) {
-		infraData, err := infraLister.Get("cluster")
+		if !infraInformer.Informer().HasSynced() {
+			return false, fmt.Errorf("infrastructures.config.openshift.io informer has not synced yet")
```

Comment:
not sure if we want to return an error here, especially it will be logged by the higher-order functions/component. After all an informer is not synced immediately.

Reply:
Would you prefer to klog.Infof and return `false, nil` instead?

Reply:
I would `return false, nil`. I would expect an informer to log failed synchronization attempts to the API server, wdyt?

Reply:
`nil` error means the function successfully retrieved the `cluster` object and determined if the topology is SNO or not. Returning `nil` when the informer cache is not synced changes the semantics.